### DISCUSSION TITLE
Design documentation properties in CodeMeta vocabulary

### DISCUSSION THREAD

**Discussion by moranegg** [codemeta/codemeta#335] [Ideas]
Following different open issues about documentation, we need a larger discussion to come to a consensus on the different properties that are representing documentation in CodeMeta.

The following can be documentation:
- build instructions
- Readme
- link to a website
- link to a repository holding the docs, a documentation module
- manual

Please go over the open issues and share your thoughts here about the next steps with the documentation property in mind.

Open issues:
- https://github.com/codemeta/codemeta/issues/245
- https://github.com/codemeta/codemeta/issues/255
- https://github.com/codemeta/codemeta/issues/247
- please propose other issues to take into account

**Comment by moranegg** [comment]
In schema.org: https://schema.org/documentation

**Comment by tpronk** [comment]
Hey! I'd like to present a software management perspective. It is inspired on eScience Center & NWO's [Practical SMP](https://zenodo.org/doi/10.5281/zenodo.7038280) and the [maSMP schema](https://github.com/zbmed-semtec/maSMPs). 

Three types of documentation I like my developers to provide (depending on what's going to happen with the software), are the following:
1. How to install the software (for using it, not developing it)
2. How to use the software
3. How contribute to it (i.e., developer documentation)

**Comment by moranegg** [comment]
Hi @tpronk that's a good point.

Do you know if in schema.org, WikiData or any other vocabulary we can find these properties?

**Reply by tpronk** [reply]
I do, but I'm also not very experienced with schemas/vocabularies/ontologies, so I asked around in my network whether someone could answer your question better than I could. If nothing happens, I'll give it a go myself

**Reply by dgarijo** [reply]
In the [software description ontology](https://w3id.org/okn/o/sd) (disclaimer, I am one of the developers) we added 
- Installation instructions
- Download instructions
- Execution instructions
- Execution command
- Usage notes
We extended schema.org and codemeta, so at the time I checked and there were no other properties on both those vocabularies to address these needs

**Comment by tpronk** [comment]
I took a quick peek at CodeMeta v3.0 from a different perspective, namely *what kind of software metadata is required/desirable to efficiently conduct a [CODECHECK](https://codecheck.org.uk/)?* [Sam Langton](https://www.samlangton.info/) provided an overview of this in his report on [Computational reproducibility at Amsterdam UMC](https://doi.org/10.5281/zenodo.15097094), page 7. All is covered by v3.0, except for the execution instructions/command. 

Note that this is still work in progress. We'll keep on conducting CODECHECKs and track what we need(ed).

**Comment by rebeccaringuette** [comment]
@moranegg potential 4.0 topic. Keep the buildInstructions term or shift to the Schema.org documentation term.

**Comment by lindangulopez** [comment]
To have properties to each type of documentation, for each instruction, is too specific by user need, but a property like  [documentation](https://schema.org/documentation) will be more useful as it collates all of these instructions. This property is however 
- new so we need to comment on this [issue](https://github.com/schemaorg/schemaorg/issues/1423)  and while this is being resolved; 
- at the same time create a new codemeta property `documentation`, of type `CreativeWork` | `Url` for v4 which resolves many of the issues around this; 
- we could deprecate `buildinstructions` before all of this.

So in this case we can kill many birds with one stone : 

- https://github.com/codemeta/codemeta/issues/416
- https://github.com/codemeta/codemeta/discussions/335
- https://github.com/codemeta/codemeta/issues/255
- https://github.com/codemeta/codemeta/issues/247
- https://github.com/codemeta/codemeta/pull/260


What do you think @dgarijo & @moranegg

**Reply by lindangulopez** [reply]
mentioned : https://github.com/schemaorg/schemaorg/issues/1423

**Reply by lindangulopez** [reply]
pinged @MatthiasWiesmann : https://github.com/codemeta/codemeta/discussions/453#discussioncomment-15095615

**Reply by lindangulopez** [reply]
see : https://github.com/schemaorg/schemaorg/issues/1423#issuecomment-3666428878

**Comment by lindangulopez** [comment]
# **ongoing discussion on the [documentation property](https://github.com/schemaorg/schemaorg/issues/1423#issuecomment-3666428878)**:

## **Use Case**:

   * **Software**: You could say `SoftwareSourceCode` "documents" a `TechArticle` which provides more detailed information about the software.
   * **Dataset**: Similarly, a dataset could be documented by a `TechArticle`, where the article explains the usage or features of the dataset.

### Concrete Example in Schema Markup:

If you have a software project and you want to link it to a guide or manual, you could structure your schema like this:

```json
{
  "@context": "http://schema.org",
  "@type": "SoftwareSourceCode",
  "name": "Example Software",
  "documentation": {
    "@type": "TechArticle",
    "headline": "Getting Started with Example Software",
    "author": {
      "@type": "Person",
      "name": "Linda Angulo Lopez"
    },
    "url": "https://example.com/guide"
  }
}
```

And for the reverse relationship, if you are looking from the `TechArticle` side, it would look something like:

```json
{
  "@context": "http://schema.org",
  "@type": "TechArticle",
  "headline": "Getting Started with Example Software",
  "documents": {
    "@type": "SoftwareSourceCode",
    "name": "Example Software"
  }
}
```

### Summary:

* **Why add `documents` as a reverse property?**
  It allows you to link back from a piece of documentation to the resource (like software or a dataset) that it describes. It makes the relationships in your data model clearer and easier to navigate, especially in more complex scenarios.

* **Thumbs up for reverse property**
  Yes, this reverse property concept is consistent with existing patterns in schema.org. It helps keep the relationship between resources clear while also making it easier to link documentation to the resource it’s describing.

**Reply by rebeccaringuette** [reply]
Makes sense to me. We need to be careful to distinguish between the reference publication and documentation, though.
It is important to consider what this would map to in DataCite since DataCite DOIs are quite popular for software. The closest resourceTypeGeneral I see is [DataPaper](https://datacite-metadata-schema.readthedocs.io/en/4.6/appendices/appendix-1/resourceTypeGeneral/#datapaper).
